packages/react-auth/src/shared/session.ts

```diff
@@ -0,0 +1,140 @@
+import { APIResponse, AuthUser, CookieOptions, Token } from "@src/types";
+import { isLocalhost } from "./utils";
+
+type CookieObj = {
+  [key: string]: string;
+};
+
+export const SESSION_DATA_KEY = "pangea-authn";
+export const DEFAULT_COOKIE_OPTIONS: CookieOptions = {
+  cookieMaxAge: 60 * 60 * 24 * 2, // 48 Hours, in seconds
+  cookieName: SESSION_DATA_KEY,
+};
+
+const BASE_COOKIE_FLAGS = "; path=/";
+
+/*
+  Session storage functions
+*/
+
+export function getStorageAPI(isUsingCookies: boolean): Storage {
+  // Depending on if we're using cookies or not, we want to use a different storage Web API
+  // When we use a cookie, use session storage
+  // When we do not use a cookie, use local storage <--- default
+  return isUsingCookies ? window.sessionStorage : window.localStorage;
+}
+
+export const getToken = (cookieName: string, storageAPI?: Storage) => {
+  if (storageAPI === undefined) {
+    const cookies = getCookies();
+
+    return cookies[cookieName];
+  }
+
+  const data = getSessionData(storageAPI);
+  return data?.active_token?.token;
+};
+
+export const getSessionData = (storageAPI = localStorage) => {
+  const data = storageAPI.getItem(SESSION_DATA_KEY);
+  const session_json = data ? JSON.parse(data) : {};
+
+  return session_json;
+};
+
+const mapTokenData = (data: any) => {
+  const tokenData: Token = (({
+    id,
+    identity,
+    token,
+    type,
+    life,
+    expire,
+    created_at,
+    scopes,
+  }) => ({ id, identity, token, type, life, expire, created_at, scopes }))(
+    data
+  );
+  return tokenData;
+};
+
+export const getUserFromResponse = (data: APIResponse): AuthUser => {
+  const activeToken = { ...data.result.active_token };
+  const refreshToken = { ...data.result.refresh_token };
```

Comment:
Do we anticipate `data.result` could ever be `undefined` at any point, like if the network request fails or something?

Reply:
Currently, this function is only called before checking if `result.active_token.token` exists, and wouldn't be called on a response that didn't return a "Success" status, so there's always a result.